src/main/java/org/sunbird/assessment/service/AssessmentServiceV2Impl.java

```diff
@@ -34,146 +39,329 @@ public class AssessmentServiceV2Impl implements AssessmentServiceV2 {
+                        for (int i = 0; i < map.size(); i++) {
+                            if (ObjectUtils.isEmpty(map.get(i))) {
+                                //Adding the not found keys as a seperate list
+                                newIdentifierList.add(identifierList.get(i));
+                            } else {
+                                //Filtering the details of the questions which are found in redis and adding them to another list
+                                questionList.add(filterQuestionMapDetail((Map<String, Object>) map.get(i)));
+                            }
+                        }
+                        //Taking the list which was formed with the not found values in Redis, we are making an internal POST call to Question List API to fetch the details
+                        if (!newIdentifierList.isEmpty()) {
+                            Map<String, Object> questionMapResponse = readQuestionDetails(newIdentifierList);
+                            if (questionMapResponse != null && !questionMapResponse.isEmpty() && Constants.OK.equalsIgnoreCase((String) questionMapResponse.get(Constants.RESPONSE_CODE))) {
+                                List<Map<String, Object>> questionMap = ((List<Map<String, Object>>) ((Map<String, Object>) questionMapResponse
+                                        .get(Constants.RESULT)).get(Constants.QUESTIONS));
+                                for (Map<String, Object> question : questionMap) {
+                                    if (!ObjectUtils.isEmpty(questionMap)) {
+                                        redisCacheMgr.putCache(Constants.QUESTION_ID + question.get(Constants.IDENTIFIER), question);
+                                        questionList.add(filterQuestionMapDetail(question));
+                                    } else {
+                                        logger.error(String.format("Failed to get Question Details for Id: %s", question.get(Constants.IDENTIFIER).toString()));
+                                    }
+                                }
+                            } else {
+                                logger.error(String.format("Failed to get Question Details from the Question List API for the IDs: %s", newIdentifierList.toString()));
+                            }
                         }
+                        if (!questionList.isEmpty())
+                            response.getResult().put(Constants.QUESTIONS, questionList);
+                        else
+                            errMsg = "The list of questions are Empty";
+                    } else {
+                        errMsg = "Please provide a valid assessment Id/Session Expired";
                     }
+                } else {
+                    errMsg = "Please provide the valid Assessment Id";
                 }
+            } else {
+                errMsg = "User Id doesn't exist! Please supply a valid auth token";
```

Comment:
can we have a validation method for this readQuestionList... the validation method should check the following... 
a) check user token is valid
b) incoming request contains assessmentId and list of questionId's
c) check the cache for readAssessmentResponse - make sure the incoming request for readQuestionList contains Id's from what we sent in readAssessmentResponse
return error response if one of the above failed...